btcec/schnorr/musig2/nonces.go

```diff
@@ -142,6 +151,14 @@ func WithCustomRand(r io.Reader) NonceGenOption {
 	}
 }
 
+// WithPublicKey is the mandatory public key that will be mixed into the nonce
```

Comment:
Not sure if it's obvious from the context or whether we should add "the mandatory public key that corresponds to the signer's private key" or something to that effect to the comment?